### PR TITLE
Fix compilation failure on Xcode 12

### PR DESCRIPTION
**Pull request by tackgyu**
This PR fixes compilation failure in building Mapper on Xcode 12 beta 2.

Reference: https://github.com/airbnb/lottie-ios/pull/1215

### CODE REVIEW

**Comment by keith** [comment]
Hrm I don't see any failures / warnings in Xcode 12 beta 1 or 2 with this, how are you building to see them?

**Comment by tackgyu** [comment]
> Hrm I don't see any failures / warnings in Xcode 12 beta 1 or 2 with this, how are you building to see them?

Sorry for misleading description. You can reproduce the failure by archiving mapper in Xcode beta 2 with target `Any iOS Device (armv7, arm64)`.

**Comment by keith** [comment]
I still don't see this on beta 2, what am I missing?

image

**Comment by tackgyu** [comment]
Ah, you can just archive the project to reproduce!

스크린샷 2020-07-14 오전 9 16 26

**Comment by keith** [comment]
ah interesting thanks, it seems it requires optimizations and linking, interesting!

File: Sources/NSDictionary+Safety.swift [hunk 1]
```diff
@@ -4,11 +4,11 @@ extension NSDictionary {
     @nonobjc
     func safeValue(forKeyPath keyPath: String) -> Any? {
         var object: Any? = self
-        var keys = keyPath.split(separator: ".").map(String.init)
+        var keys = keyPath.split(separator: ".")
 
         while !keys.isEmpty, let currentObject = object {
             let key = keys.remove(at: 0)
-            object = (currentObject as? NSDictionary)?[key]
+            object = (currentObject as? NSDictionary)?[String(key)]
```
Comment (keith) [inline Sources/NSDictionary+Safety.swift:11]: Looks like this change isn't necessary it still works?

**Review by keith — COMMENTED** [review]
This seems like only a good thing regardless of the bug, if you have a bugs.swift.org link can you add it to your description? otherwise I just have the one comment

File: Sources/NSDictionary+Safety.swift (hunk 1 above)
Comment (tackgyu) [reply Sources/NSDictionary+Safety.swift:11]: Sorry for late response. I'm not sure as the root cause of the error is unclear. I will revisit on Xcode 12 beta 3

**Comment by keith** [comment]
I've merged this with https://github.com/lyft/mapper/pull/170 and release 10.0.1 with the fix. Thanks for reporting this!

**Pull request closed without merging**